rook/scripts/test-rook.sh

```diff
@@ -0,0 +1,307 @@
+  fi
+  if [[ ! $numberOfStorageNodes = "$numberOfOSDs" ]]; then
+    echo "${numberOfOSDs}/${numberOfStorageNodes} deployments exist ❌"; FAILURES=$((FAILURES+1))
+  else
+    echo "${numberOfOSDs}/${numberOfStorageNodes} deployments exist ✔"; SUCCESSES=$((SUCCESSES+1))
+  fi
+  for osd in "${osdDeployments[@]}"; do
+    echo -n "$osd"
+    resourceReplicaCompare "Deployment" "rook-ceph" "${osd}" "${1}" "${2}"
+  done
+}
+
+# Args:
+#   1. jsonData
+#   2. cluster
+function testMons {
+  numberOfMons=$(kubectl -n rook-ceph get deployments.apps -o=name | grep -c rook-ceph-mon)
+  numberOfMonsDesired=$(yq4 ".commons.mon.count" "$CONFIG_FILE")
+  monDeployments=()
+  read -r -d '' -a monDeployments <<< "$(kubectl -n rook-ceph get deployments.apps -o=json | jq -r '.items[].metadata.name' | grep "rook-ceph-mon")"
+  if [[ -z $numberOfMonsDesired || $numberOfMonsDesired = "null" ]]; then
+    numberOfMonsDesired=$(yq4 ".cephClusterSpec.mon.count" "${here}/../helmfile.d/upstream/rook-ceph-cluster/values.yaml")
+  fi
+  if [[ ! $numberOfMons = "$numberOfMonsDesired" ]]; then
+    echo "${numberOfMons}/${numberOfMonsDesired} deployments exist ❌"; FAILURES=$((FAILURES+1))
+  else
+    echo "${numberOfMons}/${numberOfMonsDesired} deployments exist ✔"; SUCCESSES=$((SUCCESSES+1))
+  fi
+  for mon in "${monDeployments[@]}"; do
+    echo -n "$mon"
+    resourceReplicaCompare "Deployment" "rook-ceph" "${mon}" "${1}" "${2}"
+  done
+}
+
+# Args:
+#   1. jsonData
+#   2. cluster
+function testMgrs {
+  numberOfMgrs=$(kubectl -n rook-ceph get deployments.apps -o=name | grep -c rook-ceph-mgr)
+  numberOfMgrsDesired=$(yq4 ".commons.mgr.count" "$CONFIG_FILE")
+  mgrDeployments=()
+  read -r -d '' -a mgrDeployments <<< "$(kubectl -n rook-ceph get deployments.apps -o=json | jq -r '.items[].metadata.name' | grep "rook-ceph-mgr")"
+  if [[ -z "$numberOfMgrsDesired" || "$numberOfMgrsDesired" = "null" ]]; then
+    # TODO:
+    #   $(yq4 ".cephClusterSpec.mgr.count" "${here}/../helmfile.d/upstream/rook-ceph-cluster/values.yaml")
+    #   doesn't work because we set the default in
+    #   "${here}/../helmfile.d/values/cluster.yaml.gotmpl"
+    #   Is this too much or would we rather keep this up to date with the default manually since it probably won't change much?
+    #   i.e. setting numberOfMgrsDesired=1 and updating this scripts if the default ever changes
+    if [[ "${cluster}" = "sc" ]]; then
+      helmfile -e service write-values &> /dev/null
+      numberOfMgrsDesired=$(yq4 ".cephClusterSpec.mgr.count" "$(find "${here}"/../helmfile.d/helmfile-*/ -name "rook-ceph-cluster.yaml")")
+      rm -rf "$(find "${here}"/../helmfile.d/helmfile-*)"
+    elif [[ "${cluster}" = "wc" ]]; then
+      helmfile -e workload write-values &> /dev/null
+      numberOfMgrsDesired=$(yq4 ".cephClusterSpec.mgr.count" "$(find "${here}"/../helmfile.d/helmfile-*/ -name "rook-ceph-cluster.yaml")")
+      rm -rf "$(find "${here}"/../helmfile.d/helmfile-*)"
+    fi
```

Comment:
Also not sure about this, what does everyone think?

Reply:
Maybe change the `write-values` to a tmp folder instead, so like:

```bash
TMP_DIR=$(mktemp -p /tmp -d rook-ceph-test.XXXXXX)
trap "rm -rf ${TMP_DIR}" EXIT # Probably add the "append_trap" function here so nothing gets lost

helmfile -e ... write-values --output-file-template "${TMP_DIR}/{{ .State.BaseName }}-{{ .State.AbsPathSHA1 }}/{{ .Release.Name}}.yaml" &> /dev/null
```

Then we do not remove anything by mistake